src/Tools/LeanCode.CodeAnalysis/NetStandard21Compatibility/System.Range.cs

```diff
@@ -0,0 +1,133 @@
+// From https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.CoreLib/src/System/Range.cs
+// as linked by docs on 02.09.2024 with minimal changes
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+#if NETSTANDARD2_0 || NETFRAMEWORK
+using System.Numerics.Hashing;
+#endif
+
+namespace System;
+
+/// <summary>Represent a range has start and end indexes.</summary>
+/// <remarks>
+/// Range is used by the C# compiler to support the range syntax.
+/// <code>
+/// int[] someArray = new int[5] { 1, 2, 3, 4, 5 };
+/// int[] subArray1 = someArray[0..2]; // { 1, 2 }
+/// int[] subArray2 = someArray[1..^0]; // { 2, 3, 4, 5 }
+/// </code>
+/// </remarks>
+#if SYSTEM_PRIVATE_CORELIB
+public
+#else
+internal
+#endif
+readonly struct Range : IEquatable<Range>
+{
+    /// <summary>Represent the inclusive start index of the Range.</summary>
+    public Index Start { get; }
+
+    /// <summary>Represent the exclusive end index of the Range.</summary>
+    public Index End { get; }
+
+    /// <summary>Construct a Range object using the start and end indexes.</summary>
+    /// <param name="start">Represent the inclusive start index of the range.</param>
+    /// <param name="end">Represent the exclusive end index of the range.</param>
+    public Range(Index start, Index end)
+    {
+        Start = start;
+        End = end;
+    }
+
+    /// <summary>Indicates whether the current Range object is equal to another object of the same type.</summary>
+    /// <param name="value">An object to compare with this object</param>
+    public override bool Equals([NotNullWhen(true)] object? value) =>
+        value is Range r && r.Start.Equals(Start) && r.End.Equals(End);
+
+    /// <summary>Indicates whether the current Range object is equal to another Range object.</summary>
+    /// <param name="other">An object to compare with this object</param>
+    public bool Equals(Range other) => other.Start.Equals(Start) && other.End.Equals(End);
+
+    /// <summary>Returns the hash code for this instance.</summary>
+    public override int GetHashCode()
+    {
+#if (!NETSTANDARD2_0 && !NETFRAMEWORK)
+        return HashCode.Combine(Start.GetHashCode(), End.GetHashCode());
+#else
+        return HashHelpers.Combine(Start.GetHashCode(), End.GetHashCode());
+#endif
```

Comment:
Weird, https://learn.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=net-8.0#system-hashcode-combine-2(-0-1) suggests that we should be able to use it even in netstandard2.0 projects, in which case this becomes simply

```suggestion
        return HashCode.Combine(Start, End);
```

and lets you get rid of `HashHelpers` altogether.

(But you might need to add a dep on https://www.nuget.org/packages/Microsoft.Bcl.HashCode)